src/main/java/com/atlassian/db/replica/api/reason/Reason.java

```diff
@@ -32,6 +32,8 @@ private Reason(final String name, boolean isRunOnMain, boolean isWrite) {
             new ReasonBuilder("HIGH_TRANSACTION_ISOLATION_LEVEL").isRunOnMain(true).isWrite(false).build();
     public static final Reason RO_API_CALL =
             new ReasonBuilder("RO_API_CALL").isRunOnMain(false).isWrite(false).build();
+    public static final Reason REPLICA_NOT_AVAILABLE =
```

Comment:
not related to this place, more to ConnectionProvider.isReplicaAvailable(), as there is none java doc.
What's the assumption of the SPI ? does connection provider should manage the healthness of replica? and for example use it in case replica is temporary unavailable?
I see that in product we make it always true, so none fancy things overthere.
To make the impact for product would be good to change it. I wonder whether this is room for sth like CirucuitBreaker, or rather sth else?